haystack/components/generators/chat/openai_responses.py

```diff
@@ -0,0 +1,769 @@
+        :param tools_strict:
+            Whether to enable strict schema adherence for tool calls. If set to `True`, the model will follow exactly
+            the schema provided in the `parameters` field of the tool definition, but this may increase latency.
+            If set, it will override the `tools_strict` parameter set during component initialization.
+
+        :returns:
+            A dictionary with the following key:
+            - `replies`: A list containing the generated responses as ChatMessage instances.
+        """
+        # validate and select the streaming callback
+        streaming_callback = select_streaming_callback(
+            init_callback=self.streaming_callback, runtime_callback=streaming_callback, requires_async=True
+        )
+        responses: Union[AsyncStream[ResponseStreamEvent], Response]
+
+        if len(messages) == 0:
+            return {"replies": []}
+
+        api_args = self._prepare_api_call(
+            messages=messages,
+            streaming_callback=streaming_callback,
+            generation_kwargs=generation_kwargs,
+            tools=tools,
+            tools_strict=tools_strict,
+        )
+
+        openai_endpoint = api_args.pop("openai_endpoint")
+        openai_endpoint_method = getattr(self.async_client.responses, openai_endpoint)
+        responses = await openai_endpoint_method(**api_args)
+
+        if streaming_callback is not None:
+            response_output = await self._handle_async_stream_response(
+                responses,  # type: ignore
+                streaming_callback,
+            )
+
+        else:
+            assert isinstance(responses, Response), "Unexpected response type for non-streaming request."
+            response_output = [_convert_response_to_chat_message(responses)]
+        return {"replies": response_output}
+
+    def _prepare_api_call(  # noqa: PLR0913
+        self,
+        *,
+        messages: list[ChatMessage],
+        streaming_callback: Optional[StreamingCallbackT] = None,
+        generation_kwargs: Optional[dict[str, Any]] = None,
+        tools: Optional[ToolsType] = None,
+        tools_strict: Optional[bool] = None,
+    ) -> dict[str, Any]:
+        # update generation kwargs by merging with the generation kwargs passed to the run method
+        generation_kwargs = {**self.generation_kwargs, **(generation_kwargs or {})}
+
+        text_format = generation_kwargs.pop("text_format", None)
+
+        # adapt ChatMessage(s) to the format expected by the OpenAI API
+        openai_formatted_messages = [convert_message_to_responses_api_format(message) for message in messages]
+
+        tools = tools or self.tools
+        tools_strict = tools_strict if tools_strict is not None else self.tools_strict
+
+        openai_tools = {}
+        # Build tool definitions
+        if tools:
+            tool_definitions = []
+            if isinstance(tools, list) and isinstance(tools[0], dict):
```

Comment:
We use this a lot throughout the code. Let's add is_openai_tool_def function. Again I'm seeing code that allows either/or tools usage - it would be nice to equip this model with our tools and theirs. Right?